### ISSUE TITLE
Docker Desktop overwrites ECR credential helper symlink with outdated version

### ISSUE THREAD

**Issue by lachlancooper** [docker/for-mac#6986]
### Description

Docker Desktop for Mac regularly overwrites the ECR credential helper symlink, `/usr/local/bin/docker-credential-ecr-login`, pointing it at `/Applications/Docker.app/Contents/Resources/bin/docker-credential-ecr-login`, which is >18 months out of date (`v0.6.0`, released in January 2022).

This outdated version of the helper only supports the ["legacy" configuration for AWS sso](https://docs.aws.amazon.com/cli/latest/userguide/sso-configure-profile-legacy.html), so pulling images with an AWS CLI SSO profile that uses `sso-session` (the default from December 2022) doesn't work.

To get my `sso-session` profiles working I installed the current helper version [(`v0.7.1`, released May 2023)](https://github.com/awslabs/amazon-ecr-credential-helper/releases/tag/v0.7.1) via Homebrew (`brew install docker-credential-helper-ecr`) and overwrote the Docker Desktop symlink (`brew link --overwrite docker-credential-helper-ecr`).

This workaround previously worked reasonably well, though Docker Desktop would overwrite this symlink whenever it was updated. Recently Docker Desktop has also started showing a warning about "Configs changed by another App" with a prompt to overwrite its symlink.
<img width="287" alt="image" src="https://github.com/docker/for-mac/assets/660954/eacba235-b759-402c-ae9f-8c3daae83705">

This issue is nearly identical to #6180.

### Reproduce

1. Install Docker Desktop
2. Install the latest version of the ECR credential helper: `brew install docker-credential-helper-ecr`
3. Overwrite the Docker Desktop symlink: `brew unlink docker-credential-helper-ecr && brew link --overwrite docker-credential-helper-ecr`
4. View the warning appear on the Docker Desktop taskbar icon.

### Expected behavior

Docker Desktop should ship with a current version of the ECR credential helper, that at minimum supports the current default AWS CLI SSO configuration.

If this is not feasible, Docker Desktop should at least not block the user from installing their own version to work around this issue.

### docker version

```bash
Client:
 Cloud integration: v1.0.35+desktop.4
 Version:           24.0.6
 API version:       1.43
 Go version:        go1.20.7
 Git commit:        ed223bc
 Built:             Mon Sep  4 12:28:49 2023
 OS/Arch:           darwin/amd64
 Context:           desktop-linux

Server: Docker Desktop 4.23.0 (120376)
 Engine:
  Version:          24.0.6
  API version:      1.43 (minimum version 1.12)
  Go version:       go1.20.7
  Git commit:       1a79695
  Built:            Mon Sep  4 12:32:16 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.22
  GitCommit:        8165feabfdfe38c65b599c4993d227328c231fca
 runc:
  Version:          1.1.8
  GitCommit:        v1.1.8-0-g82f18fe
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
Client:
 Version:    24.0.6
 Context:    desktop-linux
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.11.2-desktop.4
    Path:     /Users/lachlan/.docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.21.0-desktop.1
    Path:     /Users/lachlan/.docker/cli-plugins/docker-compose
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     /Users/lachlan/.docker/cli-plugins/docker-dev
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.20
    Path:     /Users/lachlan/.docker/cli-plugins/docker-extension
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v0.1.0-beta.7
    Path:     /Users/lachlan/.docker/cli-plugins/docker-init
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     /Users/lachlan/.docker/cli-plugins/docker-sbom
  scan: Docker Scan (Docker Inc.)
    Version:  v0.26.0
    Path:     /Users/lachlan/.docker/cli-plugins/docker-scan
  scout: Command line tool for Docker Scout (Docker Inc.)
    Version:  0.24.1
    Path:     /Users/lachlan/.docker/cli-plugins/docker-scout

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 4
 Server Version: 24.0.6
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc io.containerd.runc.v2
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 8165feabfdfe38c65b599c4993d227328c231fca
 runc version: v1.1.8-0-g82f18fe
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
  cgroupns
 Kernel Version: 6.3.13-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 7.67GiB
 Name: docker-desktop
 ID: 5ae1d24d-5092-4241-b074-50a98665d977
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: daemon is not using the default seccomp profile
```

### Diagnostics ID

8D6DA340-C459-432B-9AFF-A80F0BBD72D4/20230921045420

**Comment by bsousaa** [comment]
We will investigate this issue. Meanwhile you can disable the Configuration Check under Advanced Settings and uncheck "Automatically check configuration"

**Comment by neonima** [comment]
Fixed with 4.28
